parsons/databases/redshift/redshift.py

```diff
@@ -420,32 +424,66 @@ def copy(self, table_obj, table_name, if_exists='fail', max_errors=0, distkey=No
                 See :ref:`parsons-table` for output options.
         """
 
-        # To Do:
-        # Auto split big files to copy more quickly
-        # Consider a json for better stability
-
-        key = self.temp_s3_copy(table_obj)
-
-        cols = None
+        # Specify the columns for a copy statement.
         if specifycols:
-            cols = table_obj.columns
+            cols = tbl.columns
+        else:
+            cols = None
 
-        try:
+        key = None
```

Comment:
Just noticed.. I don't think you would need this if you moved the `try:` down to line 460 (see comment below).